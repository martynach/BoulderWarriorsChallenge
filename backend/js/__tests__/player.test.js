const Player = require('./../Player');
const player = new Player();
const path = require("path");
const filepath = path.join(__dirname, './test_data/players.json');
player.filepath = filepath;

const fs = require('fs');
const promisify = require('./../promisify');


test('Get all players test', async () => {
    expect.assertions(1);
    const actual = await player.getAllPlayers();
    expect(actual).toMatchSnapshot();
});

test('Get all female players test', async () => {
    expect.assertions(1);
    const actual = await player.getAllPlayers('f');
    expect(actual).toMatchSnapshot();
});

test('Get all male players test', async () => {
    expect.assertions(1);
    const actual = await player.getAllPlayers('male');
    expect(actual).toMatchSnapshot();
});

test('Get all players of unknown gender test', async () => {
    expect.assertions(1);
    const actual = await player.getAllPlayers('k');
    expect(actual).toBeUndefined();
});

test('getPlayersSortedByPoints male and female test', async () => {
    expect.assertions(1);
    const actual = await player.getPlayersSortedByPoints();
    expect(actual).toMatchSnapshot();
});

test('getPlayersSortedByPoints female test', async () => {
    expect.assertions(1);
    const actual = await player.getPlayersSortedByPoints('female');
    expect(actual).toMatchSnapshot();
});

test('getPlayersSortedByPoints male test', async () => {
    expect.assertions(1);
    const actual = await player.getPlayersSortedByPoints('m');
    expect(actual).toMatchSnapshot();
});

test('getPlayersSortedByPoints unknown gender test', async () => {
    expect.assertions(1);
    const actual = await player.getPlayersSortedByPoints('fee');
    expect(actual).toBeUndefined();
});



describe('Tests for adding new player', () => {
    const tmpFilepath = path.join(__dirname, './test_data/players_tmp.json');

    beforeAll(() => {
        player.filepath = tmpFilepath;
    });

    beforeEach(async () => {
        await promisify(fs.copyFile, filepath, tmpFilepath);

    });

    afterEach(async () => {
        await promisify(fs.unlink, tmpFilepath);
    });

    afterAll(() => {
        player.filepath = filepath;
    });

    test('addNewPlayer method test for adding one player', async () => {
        expect.assertions(2);

        const newPlayer = { firstname: 'Marti', lastname: 'Chomik', gender: 'f' };
        await player.addNewPlayers([newPlayer]);

        const allPlayers = await player.getAllPlayers();
        expect(allPlayers).toMatchSnapshot();

        const tmpFileContent = await promisify(fs.readFile, tmpFilepath, 'utf8');
        expect(tmpFileContent).toMatchSnapshot();
    })

});