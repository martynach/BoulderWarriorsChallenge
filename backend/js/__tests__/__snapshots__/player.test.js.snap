// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get all female players test 1`] = `
Array [
  Object {
    "firstname": "Ola",
    "gender": "f",
    "id": 7,
    "lastname": "Chomicz",
  },
  Object {
    "firstname": "Karolina",
    "gender": "f",
    "id": 4,
    "lastname": "Kozakiewicz",
  },
  Object {
    "firstname": "Anna",
    "gender": "f",
    "id": 9,
    "lastname": "Nowak",
  },
  Object {
    "firstname": "Edyta",
    "gender": "f",
    "id": 1,
    "lastname": "Tokarczyk",
  },
  Object {
    "firstname": "Monika",
    "gender": "f",
    "id": 8,
    "lastname": "Tokarczyk",
  },
]
`;

exports[`Get all male players test 1`] = `
Array [
  Object {
    "firstname": "Wojciech",
    "gender": "m",
    "id": 3,
    "lastname": "Bańczyk",
  },
  Object {
    "firstname": "Jakub",
    "gender": "m",
    "id": 6,
    "lastname": "Bogucki",
  },
  Object {
    "firstname": "Michał",
    "gender": "m",
    "id": 2,
    "lastname": "Nowak",
  },
  Object {
    "firstname": "Damian",
    "gender": "m",
    "id": 5,
    "lastname": "Strojny",
  },
]
`;

exports[`Get all players test 1`] = `
Array [
  Object {
    "firstname": "Wojciech",
    "gender": "m",
    "id": 3,
    "lastname": "Bańczyk",
  },
  Object {
    "firstname": "Jakub",
    "gender": "m",
    "id": 6,
    "lastname": "Bogucki",
  },
  Object {
    "firstname": "Ola",
    "gender": "f",
    "id": 7,
    "lastname": "Chomicz",
  },
  Object {
    "firstname": "Karolina",
    "gender": "f",
    "id": 4,
    "lastname": "Kozakiewicz",
  },
  Object {
    "firstname": "Anna",
    "gender": "f",
    "id": 9,
    "lastname": "Nowak",
  },
  Object {
    "firstname": "Michał",
    "gender": "m",
    "id": 2,
    "lastname": "Nowak",
  },
  Object {
    "firstname": "Damian",
    "gender": "m",
    "id": 5,
    "lastname": "Strojny",
  },
  Object {
    "firstname": "Edyta",
    "gender": "f",
    "id": 1,
    "lastname": "Tokarczyk",
  },
  Object {
    "firstname": "Monika",
    "gender": "f",
    "id": 8,
    "lastname": "Tokarczyk",
  },
]
`;

exports[`Tests for adding new player addNewPlayer method test for adding one player 1`] = `
Array [
  Object {
    "firstname": "Wojciech",
    "gender": "m",
    "id": 3,
    "lastname": "Bańczyk",
  },
  Object {
    "firstname": "Jakub",
    "gender": "m",
    "id": 6,
    "lastname": "Bogucki",
  },
  Object {
    "firstname": "Ola",
    "gender": "f",
    "id": 7,
    "lastname": "Chomicz",
  },
  Object {
    "firstname": "Marti",
    "gender": "f",
    "id": 10,
    "lastname": "Chomik",
  },
  Object {
    "firstname": "Karolina",
    "gender": "f",
    "id": 4,
    "lastname": "Kozakiewicz",
  },
  Object {
    "firstname": "Anna",
    "gender": "f",
    "id": 9,
    "lastname": "Nowak",
  },
  Object {
    "firstname": "Michał",
    "gender": "m",
    "id": 2,
    "lastname": "Nowak",
  },
  Object {
    "firstname": "Damian",
    "gender": "m",
    "id": 5,
    "lastname": "Strojny",
  },
  Object {
    "firstname": "Edyta",
    "gender": "f",
    "id": 1,
    "lastname": "Tokarczyk",
  },
  Object {
    "firstname": "Monika",
    "gender": "f",
    "id": 8,
    "lastname": "Tokarczyk",
  },
]
`;

exports[`Tests for adding new player addNewPlayer method test for adding one player 2`] = `"[{\\"id\\":3,\\"firstname\\":\\"Wojciech\\",\\"lastname\\":\\"Bańczyk\\",\\"gender\\":\\"m\\",\\"top\\":5,\\"bonus\\":5},{\\"id\\":4,\\"firstname\\":\\"Karolina\\",\\"lastname\\":\\"Kozakiewicz\\",\\"gender\\":\\"f\\",\\"top\\":4,\\"bonus\\":5},{\\"id\\":2,\\"firstname\\":\\"Michał\\",\\"lastname\\":\\"Nowak\\",\\"gender\\":\\"m\\",\\"top\\":3,\\"bonus\\":5},{\\"id\\":5,\\"firstname\\":\\"Damian\\",\\"lastname\\":\\"Strojny\\",\\"gender\\":\\"m\\",\\"top\\":3,\\"bonus\\":4},{\\"id\\":1,\\"firstname\\":\\"Edyta\\",\\"lastname\\":\\"Tokarczyk\\",\\"gender\\":\\"f\\",\\"top\\":2,\\"bonus\\":3},{\\"id\\":6,\\"firstname\\":\\"Jakub\\",\\"lastname\\":\\"Bogucki\\",\\"gender\\":\\"m\\",\\"top\\":1,\\"bonus\\":2},{\\"id\\":7,\\"firstname\\":\\"Ola\\",\\"lastname\\":\\"Chomicz\\",\\"gender\\":\\"f\\",\\"top\\":0,\\"bonus\\":0},{\\"id\\":8,\\"firstname\\":\\"Monika\\",\\"lastname\\":\\"Tokarczyk\\",\\"gender\\":\\"f\\",\\"top\\":0,\\"bonus\\":0},{\\"id\\":9,\\"firstname\\":\\"Anna\\",\\"lastname\\":\\"Nowak\\",\\"gender\\":\\"f\\",\\"top\\":0,\\"bonus\\":0},{\\"id\\":10,\\"firstname\\":\\"Marti\\",\\"lastname\\":\\"Chomik\\",\\"gender\\":\\"f\\",\\"top\\":0,\\"bonus\\":0}]"`;

exports[`getPlayersSortedByPoints female test 1`] = `
Array [
  Object {
    "bonus": 5,
    "firstname": "Karolina",
    "gender": "f",
    "id": 4,
    "lastname": "Kozakiewicz",
    "top": 4,
  },
  Object {
    "bonus": 3,
    "firstname": "Edyta",
    "gender": "f",
    "id": 1,
    "lastname": "Tokarczyk",
    "top": 2,
  },
  Object {
    "bonus": 0,
    "firstname": "Ola",
    "gender": "f",
    "id": 7,
    "lastname": "Chomicz",
    "top": 0,
  },
  Object {
    "bonus": 0,
    "firstname": "Monika",
    "gender": "f",
    "id": 8,
    "lastname": "Tokarczyk",
    "top": 0,
  },
  Object {
    "bonus": 0,
    "firstname": "Anna",
    "gender": "f",
    "id": 9,
    "lastname": "Nowak",
    "top": 0,
  },
]
`;

exports[`getPlayersSortedByPoints male and female test 1`] = `
Array [
  Object {
    "bonus": 5,
    "firstname": "Wojciech",
    "gender": "m",
    "id": 3,
    "lastname": "Bańczyk",
    "top": 5,
  },
  Object {
    "bonus": 5,
    "firstname": "Karolina",
    "gender": "f",
    "id": 4,
    "lastname": "Kozakiewicz",
    "top": 4,
  },
  Object {
    "bonus": 5,
    "firstname": "Michał",
    "gender": "m",
    "id": 2,
    "lastname": "Nowak",
    "top": 3,
  },
  Object {
    "bonus": 4,
    "firstname": "Damian",
    "gender": "m",
    "id": 5,
    "lastname": "Strojny",
    "top": 3,
  },
  Object {
    "bonus": 3,
    "firstname": "Edyta",
    "gender": "f",
    "id": 1,
    "lastname": "Tokarczyk",
    "top": 2,
  },
  Object {
    "bonus": 2,
    "firstname": "Jakub",
    "gender": "m",
    "id": 6,
    "lastname": "Bogucki",
    "top": 1,
  },
  Object {
    "bonus": 0,
    "firstname": "Ola",
    "gender": "f",
    "id": 7,
    "lastname": "Chomicz",
    "top": 0,
  },
  Object {
    "bonus": 0,
    "firstname": "Monika",
    "gender": "f",
    "id": 8,
    "lastname": "Tokarczyk",
    "top": 0,
  },
  Object {
    "bonus": 0,
    "firstname": "Anna",
    "gender": "f",
    "id": 9,
    "lastname": "Nowak",
    "top": 0,
  },
]
`;

exports[`getPlayersSortedByPoints male test 1`] = `
Array [
  Object {
    "bonus": 5,
    "firstname": "Wojciech",
    "gender": "m",
    "id": 3,
    "lastname": "Bańczyk",
    "top": 5,
  },
  Object {
    "bonus": 5,
    "firstname": "Michał",
    "gender": "m",
    "id": 2,
    "lastname": "Nowak",
    "top": 3,
  },
  Object {
    "bonus": 4,
    "firstname": "Damian",
    "gender": "m",
    "id": 5,
    "lastname": "Strojny",
    "top": 3,
  },
  Object {
    "bonus": 2,
    "firstname": "Jakub",
    "gender": "m",
    "id": 6,
    "lastname": "Bogucki",
    "top": 1,
  },
]
`;
